cloudflare.go

```diff
@@ -40,7 +40,7 @@ func (api *API) makeRequest(method, uri string, params interface{}) ([]byte, err
 	// Replace nil with a JSON object if needed
 	var reqBody io.Reader
 	switch method {
-	case "GET", "DELETE":
+	case "GET":
```

Comment:
I think this will break `DeleteDNSRecord`.

The `default` case may need an additional check if `params == nil` and set `reqBody`, else do the rest.